### PR TITLE
Pail migrate job. TODO: Test run

### PR DESCRIPTION
**Pull request by gsriram7**

### CODE REVIEW

**Comment by ashwanthkumar** [comment]
@gsriram7 The build failed on the PR.

File: dfs-datastores/src/main/scala/com/indix/pail/PailMigrate.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      println("output_dir - Destination pail dir to which records need to be written to")
+      println("pail_spec_class_fqcn - Fully qualified class name of output PailStructure. This will be used to decide the partitioning scheme")
+      println("keep_source_files - set this to true to keep the source files. This is optional, Defaults to false, which means, source files will be removed if the job succeeds")
+      System.exit(1)
+    }
+
+    /*Input Pail location to pick up and migrate
+    * Input paths will be deleted unless `keepSourceFiles` flag is set
+    * */
+    val inputDir = args(0)
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/scala/com/indix/pail/PailMigrate.scala]: Does it help to add `scalding-args` dependency to the project for the args parsing? That way we don't need these long comments for each arg.

File: dfs-datastores/src/main/scala/com/indix/pail/PailMigrate.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import com.indix.pail.PailMigrate.PailMigrateMapper
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.io.{BytesWritable, Text}
+import org.apache.hadoop.mapred._
+import org.apache.hadoop.mapreduce.Job
+import org.apache.hadoop.util.Tool
+import org.apache.log4j.Logger
+
+class PailMigrate extends Tool {
```
Comment (ashwanthkumar) [inline dfs-datastores/src/main/scala/com/indix/pail/PailMigrate.scala]: Any reason why this is not a Scalding job but a vanilla Hadoop MR job?

**Comment by gsriram7** [comment]
we want to merge it to develop branch and test along with PailConsolidate thats why we merged it

**Comment by ashwanthkumar** [comment]
@gsriram7 This PR is not yet merged right?

**Comment by gsriram7** [comment]
no not yet. we merged it in our local and resolving compilation issues

**Comment by sriram-r** [comment]
​Ashwanth - We could write it as a scalding job too, but, that would mean,
unnecessary dependency on either scalding-ext or scalding itself.
The trouble with having scalding-ext ​is circular dependency (because, the
right home for this is dfs-datastores not scalding-ext). The reason for
avoiding scalding is the dependency size itself is too big for just this
small feature.

Considering all this, I felt, it's best to keep it as simple vanilla MR.
Thoughts?

**Comment by ashwanthkumar** [comment]
@sriram-r Makes sense, I'm sorry I forgot we don't have scalding dependency in here. Do you think we should have these tools in a separate module outside dfs-datastores? Reasons being
1. Keeps the core dfs-datastores small (without Scala deps)
2. Helps clearly differentiate between the core and tools also while publishing (in reference to #7 )

**Comment by sriram-r** [comment]
I agree. But, Archiver technically should've been part of core. We can
rewrite it in Java, if scala deps is a problem.

**Pull request merged**